File: languages/csharp/01-foundations/strings/example/string-clean-and-tokenize.cs
// This helper example focuses on isolating string cleanup before the main example combines the steps.

// This extra example extends strings with text normalization and token extraction.

using System;
using System.Collections.Generic;

// Keep this helper separate so the main example can focus on the larger idea without extra noise.
class Program
{
    static void Main()
    {
        Console.Write("Enter a sentence: ");
        string line = Console.ReadLine() ?? string.Empty;

        char[] cleanedChars = new char[line.Length];
        for (int index = 0; index < line.Length; index++)
        {
            char current = line[index];
            cleanedChars[index] = char.IsLetterOrDigit(current)
                ? char.ToLowerInvariant(current)
                : ' ';
        }

        string cleaned = new string(cleanedChars);
        List<string> words = new List<string>(
            cleaned.Split(' ', StringSplitOptions.RemoveEmptyEntries)
        );

        Console.WriteLine($"Normalized text: {cleaned}");
        Console.WriteLine($"Tokens ({words.Count}):");
        foreach (string word in words)
        {
            Console.WriteLine($"- {word}");
        }
    }
}

File: languages/python/01-foundations/strings/example/string-clean-and-tokenize.py
# This extra example extends strings with text normalization and token extraction.
# Example purpose: separate character cleanup from token building.

line = input("Enter a sentence: ")

cleaned_chars = []
for char in line:
    if char.isalnum():
        cleaned_chars.append(char.lower())
    else:
        cleaned_chars.append(" ")

cleaned = "".join(cleaned_chars)

words = []
current_word = []
for char in cleaned:
    if char == " ":
        if current_word:
            words.append("".join(current_word))
            current_word = []
    else:
        current_word.append(char)

if current_word:
    words.append("".join(current_word))

print(f"Normalized text: {cleaned}")
print(f"Tokens ({len(words)}):")
for word in words:
    print(f"- {word}")
